src/Fixer/ControlStructure/NoUnneededControlParenthesesFixer.php

```diff
@@ -115,16 +102,55 @@ final class NoUnneededControlParenthesesFixer extends AbstractFixer implements C
         T_YIELD_FROM => 'yield_from',
     ];
 
-    // handled by the `include` rule
+    /**
+     * handled by the `include` rule.
+     *
+     * @var list<int>
+     */
     private const TOKEN_TYPE_NO_CONFIG = [
         T_REQUIRE,
         T_REQUIRE_ONCE,
         T_INCLUDE,
         T_INCLUDE_ONCE,
     ];
 
+    /**
+     * @var list<list<int>|string>
```

Comment:
```suggestion
     * @var list<list<int>|string>
     * @readonly
```

Reply:
It's not read-only, elements from `Token::getObjectOperatorKinds()` are appended after the initial assignment.

Reply:
Potato potato isn't it? by the time we can change this annotation `@readonly` into a real `readonly property` we no longer need it as a `property` as it can and should become a `const`.

Reply:
But for now it's not read-only, even PHPStan reports this.

Reply:
it cannot be const if it's executed in runtime
eg this will crash
`const foo = [1, 2, 3] + createArray()`
but this will not
`readonly foo = [1, 2, 3] + createArray()`

Reply:
@jrmajor , any better annotation to mark it as readonly before 8.x?
8.x allows for readonly property initialised in constructor, in PHP and other languages
eg https://3v4l.org/bSBJA

Reply:
>it cannot be const if it's executed in runtime

my point is/was that this is no longer the case when `readonly` is available as at that point there is no need to have a `readonly` attribute as it can be a `const`, so no runtime needed.

Reply:
There was no readonly attribute in past in PHP at all. and the CI was still passing.
it's nothing required, but I also see no harm in adding it and CI is green for it. And it's future-ready for us when we will have https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/6640 in place. Feel free to ignore this proposal and I will raise it as separated PR, after this one will get merged.

> My point is/was that this is no longer the case when readonly is available as at that point there is no need to have a readonly attribute as it can be a const, so no runtime needed.

I would kind-of compare it to `@const` or `@private` annotations we were using when we were not yet relying on language built-in function, and we still wanted to indicate it.

Reply:
I didn't get it to work with the CI, I try again when I have the time :+1:

Reply:
I don't see a way to do this with `@readonly` without SCA tripping. I really don't see the need here either. It is becoming really frustrating getting anything changed here in the project.